src/scikit_build_core/ast/tokenizer.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+import dataclasses
+import enum
+import re
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from .._logging import rich_print
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+
+__all__ = ["Token", "TokenType", "tokenize"]
+
+
+def __dir__() -> list[str]:
+    return __all__
+
+
+TOKEN_EXPRS = {
+    "BRACKET_COMMENT": r"\s*#\s*\[(?P<bc1>=*)\[(?s:.)*?\](?P=bc1)\]",
+    "COMMENT": r"#.*$",
+    "QUOTED": r'"(?:\\(?s:.)|[^"\\])*?"',
+    "BRACKET_QUOTE": r"\[(?P<bq1>=*)\[(?s:.)*?\](?P=bq1)\]",
+    "OPEN_PAREN": r"\(",
+    "CLOSE_PAREN": r"\)",
+    "LEGACY": r'\b\w+=[^\s"()$\\]*(?:"[^"\\]*"[^\s"()$\\]*)*|"(?:[^"\\]*(?:\\.[^"\\]*)*)*"',
```

Comment:
Didn't know about this legacy part. Couldn't find the documentation of it and its depreciation/removal. Is it compatible with the minimum CMake that scikit-build-core requires?

Just for navigability, can the bracket section (and maybe other parts) be extracted out of the dict and introduced in as raw f-strings?

Reply:
https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#unquoted-argument

I think it's in the "not really recommended for new code" stage.

I've downloaded every CMakeLists.txt from PyPI, so I have 35,000 test cases. I haven't quite finished, but it's passing about 10,000 before it hits its first failure currently. I sometimes have to skip a file because it's not valid. (One literally was a C++ file named CMakeLists.txt, for example).

Reply:
Found one more small issue, now passes 32,000+ files (parsing the whole thing without crashing). Had to skip 20 files, mostly templates, or other clearly broken files.